cilium/cmd/cleanup.go

```diff
@@ -56,6 +59,7 @@ const (
 
 func init() {
 	rootCmd.AddCommand(cleanupCmd)
+	cleanupCmd.Flags().BoolVar(&bpfStateOnly, "bpf-state-only", false, "Remove only BPF state")
```

Comment:
As a matter of style, naming an argument like "--foo-only" means that if we need to add other conditional sets of cleanups in future, then it becomes possible specify "--foo-only --bar-only". In that event, it's confusing whether that's a no-op or whether it's "only" both options.

Could we do something like have a required argument instead, either `cilium cleanup bpf` or `cilium cleanup all`? This would also allow the below code to be something like `if cleanupBPF { ... }` with something at the start that just says `if all { cleanupBPF = true }`.

Reply:
Yes, this makes more sense! Do you see any potential downsides to having them as flags instead of arguments, so that e.g `cilium cleanup bpf` becomes `cilium cleanup --bpf`? Since this will enable the approach that @aanm suggests of binding them to env vars instead of having checks in the YAML.

Reply:
That seems reasonable. My only question would be what does `cilium cleanup` do vs. `cilium cleanup --all`? Would they be the same? (I guess this would retain the previous behaviour, for whatever that is worth)

Reply:
Right, it doesn't seem to fit nicely with this particular case. Intuitively, I could see `cilium cleanup` being the same as `--all` but also as a no-op - essentially requiring a flag to be used with the command. I'll update this PR having it preserve the previous behavior and we can see what works and what doesn't.